src/configurationProvider.ts

```diff
@@ -195,16 +194,115 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         return !Array.isArray(configItems) || !configItems.length;
     }
 
-    private async chooseMainClass(folder: vscode.WorkspaceFolder | undefined): Promise<IMainClassOption | undefined> {
-        const res = await resolveMainClass(folder ? folder.uri : undefined);
+    private async quickFixLaunchConfig(folder: vscode.Uri | undefined, config: vscode.DebugConfiguration):
+        Promise<IMainClassOption | undefined> {
+        if (!config.mainClass) {
+            return this.chooseMainClass(folder);
+        }
+
+        const validationResponse = await validateLaunchConfig(folder, config.mainClass, config.projectName);
+        if (validationResponse.mainClass.passed && validationResponse.projectName.passed) {
+            return {
+                mainClass: config.mainClass,
+                projectName: config.projectName,
+            };
+        } else {
+            const errors: string[] = [];
+            if (!validationResponse.mainClass.passed) {
+                errors.push(String(validationResponse.mainClass.message));
+            }
+
+            if (!validationResponse.projectName.passed) {
+                errors.push(String(validationResponse.projectName.message));
+            }
+
+            if (validationResponse.proposals && validationResponse.proposals.length) {
+                const answer = await utility.showErrorMessageWithTroubleshooting({
+                    message: errors.join("\n"),
+                    type: Type.USAGEERROR,
+                }, "Quick Fix");
+                if (answer === "Quick Fix") {
+                    const quickFix: IMainClassOption = await this.showMainClassQuickPick(validationResponse.proposals,
```

Comment:
What if user cancels and returns nothing?

Reply:
If the user chooses to cancel the fix QuickPick box, then the debugger exits.